python/ray/train/v2/tests/test_data_parallel_trainer.py

```diff
@@ -141,6 +141,55 @@ def train_fn():
         assert tmp_path.joinpath("validate", str(rank)).exists()
 
 
+def run_process_for_get_all_reported_checkpoints(tmp_path):
+    # Lives outside test_get_all_reported_checkpoints because cannot pickle nested functions.
+
+    # Needed to reuse current ray cluster.
+    ray.init(address="auto")
+
+    def train_fn():
+        if ray.train.get_context().get_world_rank() == 0:
+            with create_dict_checkpoint({}) as checkpoint:
+                ray.train.report(metrics={}, checkpoint=checkpoint)
+            assert len(ray.train.get_all_reported_checkpoints()) == 1
+        else:
+            signal_actor = ray.get_actor(
+                "signal_actor", namespace="test_report_get_all_reported_checkpoints"
+            )
+            ray.get(signal_actor.wait.remote())
+            ray.train.report(metrics={}, checkpoint=None)
+
+    trainer = DataParallelTrainer(
+        train_fn,
+        scaling_config=ScalingConfig(num_workers=2),
+        run_config=RunConfig(storage_path=str(tmp_path)),
+    )
+    trainer.fit()
+
+
+def test_report_get_all_reported_checkpoints(tmp_path):
+    """Check that get_all_reported_checkpoints waits until all workers have reported."""
+    SignalActor = create_remote_signal_actor(ray)
+    signal_actor = SignalActor.options(
+        name="signal_actor", namespace="test_report_get_all_reported_checkpoints"
+    ).remote()
```

Comment:
do we need this external signal actor? can we simplify the test to just test with a train_fn and some rank-specific logic? for example, that `report, get, report` vs `report, report, get` that I had above.

Reply:
changed.